### DISCUSSION TITLE
Interoperability Windows Transport Mode?

### DISCUSSION THREAD

**Discussion by agowa** [strongswan/strongswan#1308] [Q&A]
Currently, the documentation only shows how to use the VPN client in windows to connect to strongswan (I.E. only IPSec tunnel mode).

Did anyone manage to get the transport mode using Connection Security Rules of the Windows Filtering Platform (wf.msc) working?

Reference documentation from Microsoft:
* https://learn.microsoft.com/en-us/training/modules/explore-connection-security-rules/?WT.mc_id=modinfra-0000-orthomas
* https://learn.microsoft.com/en-us/windows/security/threat-protection/windows-firewall/securing-end-to-end-ipsec-connections-by-using-ikev2?source=recommendations
* https://learn.microsoft.com/en-us/windows/security/threat-protection/windows-firewall/windows-firewall-with-advanced-security-administration-with-windows-powershell#deploy-secure-firewall-rules-with-ipsec
* https://learn.microsoft.com/en-us/windows/security/threat-protection/windows-firewall/windows-firewall-with-advanced-security
* (3rd party how-to using the GUI) https://anthonyfontanez.com/index.php/2021/09/16/windows-firewall-part-4-identity-based-access-control-via-kerberos/

**Comment by testcaoy7** [comment]
I tried this morning but with no success. 
Scenario:
LAN Address Space: 7.7.0.0/16

strongSwan Server: 7.7.7.94
strongSwan Version: 5.9.5
Linux Version: Ubuntu 22.04

Windows Client: Any address located in 7.7.0.0/16 (currently 7.7.7.57)
Windows Version: Windows 10 Enterprise 21H2 64bit

Useage:
I started a socks proxy in strongSwan server (7.7.7.94), the proxy ran at TCP port 10080. 
The transport mode should secure nothing but TCP ingress for port 10080 and TCP egress for %any.

Problem:
I do not know how to start an ipsec SA from Windows side. I configured Group Policy and Advanced Firewall but only to find the proxy is not working. (I configured the Microsoft Edge browser to use socks proxy located at 7.7.7.94)

I ran `ipsec statusall` and the output shown no SA was established.
output:
`
Listening IP addresses:
  7.7.7.94
Connections:
        test:  7.7.7.94...%any  IKEv2, dpddelay=20s
        test:   local:  uses pre-shared key authentication
        test:   remote: uses pre-shared key authentication
     default:   child:  7.7.7.94/32[tcp/amanda] === dynamic[tcp] TRANSPORT, dpdaction=clear
Routed Connections:
     default{1}:  ROUTED, TRANSPORT, reqid 1
     default{1}:   7.7.7.94/32[tcp/amanda] === 0.0.0.0/0[tcp]
Security Associations (0 up, 0 connecting):
  none
`

I have attached my linux.swanctl.conf. 
[linux.swanctl.txt](https://github.com/strongswan/strongswan/files/9603595/linux.swanctl.txt)
